src/simulation.md

```diff
@@ -2,3 +2,202 @@
+the type of door (hinged, double_hinged, sliding, double_sliding) and its range
+of motion (for hinged doors).
+
+
+![Figure X](images/door_traffic_editor.png)
+
+The `building_map_generator gazebo` script parses a `.building.yaml` file for
+any doors and automatically generates an sdf sub-element with links and joints
+required for the door along with a configured plugin. The sdf sub-element
+generated for the door in the figure above is presented below.
+
+```xml
+<model name="coe_door">
+  <pose>8.077686357313898 -5.898342045416362 0.0 0 0 1.1560010438234292</pose>
+  <plugin filename="libdoor.so" name="door">
+    <v_max_door>0.5</v_max_door>
+    <a_max_door>0.3</a_max_door>
+    <a_nom_door>0.15</a_nom_door>
+    <dx_min_door>0.01</dx_min_door>
+    <f_max_door>500.0</f_max_door>
+    <door left_joint_name="left_joint" name="coe_door" right_joint_name="empty_joint" type="SwingDoor" />
+  </plugin>
+  <link name="left">
+    <pose>0 0 1.11 0 0 0</pose>
+    <visual name="left">
+      <material>
+        <ambient>120 60 0 0.6</ambient>
+        <diffuse>120 60 0 0.6</diffuse>
+      </material>
+      <geometry>
+        <box>
+          <size>0.8766026166317483 0.03 2.2</size>
+        </box>
+      </geometry>
+    </visual>
+    <collision name="left">
+      <surface>
+        <contact>
+          <collide_bitmask>0x02</collide_bitmask>
+        </contact>
+      </surface>
+      <geometry>
+        <box>
+          <size>0.8766026166317483 0.03 2.2</size>
+        </box>
+      </geometry>
+    </collision>
+    <inertial>
+      <mass>50.0</mass>
+      <inertia>
+        <ixx>20.17041666666667</ixx>
+        <iyy>23.36846728119012</iyy>
+        <izz>3.20555061452345</izz>
+      </inertia>
+    </inertial>
+  </link>
+  <joint name="left_joint" type="revolute">
+    <parent>world</parent>
+    <child>left</child>
+    <axis>
+      <xyz>0 0 1</xyz>
+      <limit>
+        <lower>-1.57</lower>
+        <upper>0</upper>
+      </limit>
+    </axis>
+    <pose>0.44330130831587417 0 0 0 0 0</pose>
+  </joint>
+</model>
+```
+
+The door [plugin](https://github.com/osrf/traffic_editor/blob/master/building_gazebo_plugins/src/door.cpp) responds to `DoorRequest` messages with `door_name` matching its `model name` sdf tag. These messages are published over the `/door_requests` topic. The plugin is agnostic of the type of door defined and relies on the `left_joint_name` and `right_joint_name` parameters to determine which joints to actuate during open and close motions. During these motions, the joints are commanded to their appropriate limits which are specified in the parent element. The joint motions follow similar acceleration and deceleration profiles as in the `slotcar`. 
+ 
+
+#### Lifts
+
+
+#### Dispensers
+A common scenario of interest is that of robots performing deliveries where a payload is loaded onto the robot at one location and the unloaded at another.
```

Comment:
```suggestion
A common scenario of interest is that of robots performing deliveries where a payload is loaded onto a robot at one location and then unloaded at another. These load and unload tasks can involve robot manipulator actions as well.
```

Reply:
The comment seems to be on an outdated version.